File: poc-platform/backend/app/schemas/recycle_bin.py
from datetime import datetime
from pydantic import BaseModel


class RecycleBinItem(BaseModel):
    id: str
    type: str
    name: str
    deleted_at: datetime
    deleted_by: str | None
    extra: dict | None = None


class RecycleBinResponse(BaseModel):
    items: list[RecycleBinItem]
    total: int

File: poc-platform/frontend/src/api/recycle-bin.ts
import client from './client';

export interface RecycleBinItem {
  id: string;
  type: 'project' | 'dashboard' | 'log' | 'option' | 'diagram' | 'chart' | 'sop_document' | 'test_case' | 'script' | 'sop_category';
  name: string;
  deleted_at: string;
  deleted_by: string | null;
  extra: Record<string, any> | null;
}

export interface RecycleBinResponse {
  items: RecycleBinItem[];
  total: number;
}

export function getRecycleBin() {
  return client.get<RecycleBinResponse>('/recycle-bin/');
}

export function restoreItem(type: string, id: string) {
  return client.post(`/recycle-bin/restore/${type}/${id}`);
}

export function permanentDeleteItem(type: string, id: string) {
  return client.delete(`/recycle-bin/permanent/${type}/${id}`);
}
